### PR TITLE
Add size and multiple to select explainer

### PR DESCRIPTION
**Pull request by josepharhar**
This follow the resolution in
https://github.com/openui/open-ui/issues/977

### CODE REVIEW

**Comment by josepharhar** [comment]
@mfreed7

File: site/src/pages/components/selectlist.mdx [hunk 1]
```diff
@@ -487,6 +490,10 @@ When the listbox is open:
 * The down arrow key moves focus forwards in the list of options and changes the selected value to the newly selected option.
 * The left and right arrow keys do nothing.
 
+## `multiple` and `size` attributes
+
+The HTML parser will not allow `<button>` or `<datalist>` children when the `multiple` or `size` attributes are present on `<select>`. This will ensure that the old rendering behavior of `multiple` and `size` is used. If the author adds a `<datalist>` or `<button>` via script to a `<select multiple>` or `<select size>`, the old rendering will still be used.In the future, we would like to upgrade `<select multiple>` to become stylable, at which time we will change the HTML parser again to allow `<button>` and `<datalist>` children. This also allows feature detection to know whether `<select multiple>` improvements are ready to be used. More detail in [this issue](https://github.com/openui/open-ui/issues/977#issuecomment-1910874971).
```
Comment (mfreed7) [inline site/src/pages/components/selectlist.mdx:495]: Needs a space between '.' and 'In the future'

**Review by mfreed7 — APPROVED** [review]
LGTM

File: site/src/pages/components/selectlist.mdx (hunk 1 above)
Comment (josepharhar) [reply site/src/pages/components/selectlist.mdx]: done

**Pull request merged**